src/mp-instance.js

```diff
@@ -84,6 +90,8 @@ export default function mParticleInstance(instanceName) {
     this._Identity = new Identity(this);
     this.Identity = this._Identity.IdentityAPI;
     this.generateHash = this._Helpers.generateHash;
+
+    // QUESTION: Why do we not init store here instead of persistence?
```

Comment:
Not sure I follow the question.  Persistence is not being initialized here.  the getDeviceId method lives on persistence (maybe it shouldn't), and we are exposing that method to the user.  Here's a great example of where the persistence module needs to remain initialized despite turning saving to persistence off.

Reply:
My thought is that the Device ID specifically should be part of the Store (or at the very least independent of persistence) since it we are simply generating a random GUID if persistence is unavailable. I think this lines up with the idea of moving the Store "in front of" Persistence, so that we treat Store as the record of truth.